### ISSUE TITLE
Core: Use either "modified" or "lastmodified" for table fields

### ISSUE THREAD

**Issue by muratpurc** [CONTENIDO/CONTENIDO#107]
We currently have two table field names for storing the last modified time of an entry, `modified` and `lastmodified`, see `con_art_lang.lastmodified` and `communications.modified`. This should be unified.

Access to old names of renamed table fields can be handled in the `get()` function of appropriate `Item` classes. But this won't work for the case, when a module or plugin self-created SQL statements.

Tags:
database, setup, update, upgrade

**Comment by gonatan** [comment]
I absolutly advocate this proposal in terms of uniformity, but wouldn't be the risc to break third-party modules or plugins to high?

**Comment by muratpurc** [comment]
@gonatan yes, this is a big problem. normalizing the names of the date fields wouldn't be dowswards compatible since not everybody uses the Item and ItemCollection classes to access the database tables.

**Comment by Faar400** [comment]
> ... since not everybody uses the Item and ItemCollection classes to access the database tables.

It's funny because it's true.
I think, most of users modules won't use the Item and Item Collection classes.

**Comment by dceonline** [comment]
Go and mark them first as deprecated for a future version, hold them and write this into version doc of current version and change fields later for that future version.
Publish this in the forum and give module developers time to change their modules to use Item and ItemCollection class.

**Comment by gonatan** [comment]
@dceonline Sounds like you are volunteering for this task. Thumbs up!

**Comment by dceonline** [comment]
Not volunteering, i just do it for the next release

@muratpurc please add this to our release documentation

Also we need a list of tables and fields we will change in future versions. And we also should another time ask developers to use genericdb classes instead of using direct access to db

**Comment by dceonline** [comment]
Maybe it's a good idea to add some default DateTime fields directly with genericDB, so we use it with all tables using genericDB?
What about
- created - CURRENT_TIMESTAMP 
- modified - CURRENT_TIMESTAMP, ON UPDATE CURRENT_TIMESTAMP
- ?

and fields for user
- createdby - id of user
- modifiedby - id of user

**Comment by muratpurc** [comment]
@dceonline I suggest that we implement these changes only in a new major version, as they would significantly alter the existing interfaces.
